### PR TITLE
[5.6] Add truncate option to seed command

### PR DESCRIPTION
**Pull request by ahmed-aliraqi**
if you want to truncate all tables when seeding you should to run this command.
```
php artisan db:seed --trancate
```

### CODE REVIEW

File: src/Illuminate/Database/Console/Seeds/SeedCommand.php [hunk 1]
```diff
@@ -36,7 +38,7 @@ class SeedCommand extends Command
     /**
      * Create a new database seed command instance.
      *
-     * @param  \Illuminate\Database\ConnectionResolverInterface  $resolver
+     * @param  \Illuminate\Database\ConnectionResolverInterface $resolver
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Seeds/SeedCommand.php]: please don\t remove that extra space

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Database/Console/Seeds/SeedCommand.php [hunk 2]
```diff
@@ -88,6 +94,29 @@ protected function getDatabase()
...
+     */
+    protected function truncateTables()
+    {
+        DB::statement('SET FOREIGN_KEY_CHECKS=0;');
+
+        $tableNames = Schema::getConnection()->getDoctrineSchemaManager()->listTableNames();
+        $this->info('Truncate tables...');
+        foreach ($tableNames as $name) {
+            //don't truncate migrations
+            if ($name == 'migrations') {
```
Comment (TBlindaruk) [inline src/Illuminate/Database/Console/Seeds/SeedCommand.php]: Please change to 'migrations' === $name

**Comment by browner12** [comment]
currently I just have a truncate command at the top of my seeders.

```php
User::truncate()
```

Is there a use case for this new feature, that isn't handled by just doing that?

File: src/Illuminate/Database/Console/Seeds/SeedCommand.php [hunk 3]
```diff
@@ -88,6 +94,29 @@ protected function getDatabase()
         return $database ?: $this->laravel['config']['database.default'];
     }
 
+    /**
+     * Truncate all tables in database.
+     *
+     * @return void
+     */
+    protected function truncateTables()
+    {
```
Comment (TBlindaruk) [inline src/Illuminate/Database/Console/Seeds/SeedCommand.php]: please add a return type to the method 
```PHP
 protected function truncateTables() : void
```

**Review by TBlindaruk — CHANGES_REQUESTED** [review]

**Comment by ahmed-aliraqi** [comment]
@browner12 It will truncate all tables at once just like `migrate:fresh` and it optional feature

File: src/Illuminate/Database/Console/Seeds/SeedCommand.php (hunk 3 above)
Comment (arjasco) [reply src/Illuminate/Database/Console/Seeds/SeedCommand.php]: @TBlindaruk I'm pretty sure these return types are not enforced anywhere within the framework, regardless of 
 5.6 requiring PHP 7.1+

**Comment by arjasco** [comment]
Only issue here is running a single seeder. Maybe you only want to truncate that 1 table and re-seed it with some new data so you run the below:

`php artisan db:seed --class=UsersTableSeeder --truncate`

This command would truncate your entire database and then only seed your user table. As @browner12 pointed out, this is why calling truncate on your Model / Table within your seeder is generally the accepted way.

File: src/Illuminate/Database/Console/Seeds/SeedCommand.php (hunk 3 above)
Comment (GrahamCampbell) [reply src/Illuminate/Database/Console/Seeds/SeedCommand.php]: Yes, laravel doesn't use scalar types or return types.

File: src/Illuminate/Database/Console/Seeds/SeedCommand.php [hunk 4]
```diff
@@ -88,6 +94,29 @@ protected function getDatabase()
...
+        DB::statement('SET FOREIGN_KEY_CHECKS=0;');
```
Comment (sisve) [inline src/Illuminate/Database/Console/Seeds/SeedCommand.php]: This looks like Mysql. Have you verified that this works on SQL Server too?

**Comment by sisve** [comment]
Wouldn't this be identical to `php artisan migrate:fresh --seed`?

**Pull request closed without merging**